src/Atmos/Model/AtmosModel.jl

```diff
@@ -213,49 +302,38 @@ Constructor for `AtmosModel` (where `AtmosModel <: BalanceLaw`).
 """
 function AtmosModel{FT}(
     orientation::Orientation,
-    param_set::AbstractParameterSet;
+    physics::AtmosPhysics;
     init_state_prognostic = nothing,
     problem = AtmosProblem(init_state_prognostic = init_state_prognostic),
-    energy = EnergyModel(),
-    ref_state = HydrostaticState(DecayingTemperatureProfile{FT}(param_set),),
-    turbulence = SmagorinskyLilly{FT}(C_smag(param_set)),
-    turbconv = NoTurbConv(),
-    hyperdiffusion = NoHyperDiffusion(),
-    viscoussponge = NoViscousSponge(),
-    moisture = EquilMoist{FT}(),
-    precipitation = NoPrecipitation(),
-    radiation = NoRadiation(),
     source = (
         Gravity(),
         Coriolis(),
         GeostrophicForcing(FT, 7.62e-5, 0, 0),
-        turbconv_sources(turbconv)...,
+        turbconv_sources(physics.turbconv)...,
     ),
-    tracers = NoTracers(),
-    lsforcing = NoLSForcing(),
-    compressibility = Compressible(),
     data_config = nothing,
 ) where {FT <: AbstractFloat}
     @assert !any(isa.(source, Tuple))
 
     atmos = (
-        param_set,
+        physics,
         problem,
         orientation,
-        energy,
-        ref_state,
-        turbulence,
-        turbconv,
-        hyperdiffusion,
-        viscoussponge,
-        moisture,
-        precipitation,
-        radiation,
         source,
-        tracers,
-        lsforcing,
-        compressibility,
         data_config,
+        physics.param_set,
+        physics.energy,
+        physics.ref_state,
+        physics.turbulence,
+        physics.turbconv,
+        physics.hyperdiffusion,
+        physics.viscoussponge,
+        physics.moisture,
+        physics.precipitation,
+        physics.radiation,
+        physics.tracers,
+        physics.lsforcing,
+        physics.compressibility,
```

Comment:
I suppose this is the part that will change in the future?

Reply:
Yes

Reply:
We could get rid of the duplicate fields, and define
```julia
const PhysicsSyms = fieldnames(AtmosPhysics)
const PhysicsProps = Union{typeof.(Val.(PhysicsSyms))...}
unval(::Type{Val{S}}) where {S} = S
Base.getproperty(atmos::AtmosModel, sym::Symbol) = getproperty(atmos, Val(sym))
Base.getproperty(atmos::AtmosModel, ::T) where {T <: PhysicsProps} = getfield(atmos.physics, unval(T))
Base.getproperty(atmos::AtmosModel, ::T) where {T} = getfield(atmos, unval(T))
```
but I think there are reservations about overloading `Base.getproperty(::AtmosModel, ::Symbol)`. @simonbyrne ?

Reply:
Actually, that's not GPU compatible, but this works:
```julia
function Base.getproperty(atmos::AtmosModel, sym::Symbol)
    if sym === :problem
        return Base.getfield(atmos, sym)
    elseif sym === :orientation
        return Base.getfield(atmos, sym)
    elseif sym === :source
        return Base.getfield(atmos, sym)
    elseif sym === :data_config
        return Base.getfield(atmos, sym)
    elseif sym === :physics
        return Base.getfield(atmos, sym)
    else
        physics = Base.getfield(atmos, :physics)
        return Base.getproperty(physics, sym)
    end
end
```

Reply:
I'm still not quite happy with the duplication, nor with this solution. I've opened #2105 (and we'll need several similar PRs for the other properties) in hopes to avoid the duplication and not need this hack.

Reply:
This has now been resolved as we have moved all of the properties into `AtmosPhysics` without duplicating.